packages/integrations/vercel/src/serverless/middleware.ts

```diff
@@ -36,17 +31,26 @@ export async function generateEdgeMiddleware(
 		platform: 'browser',
 		// https://runtime-keys.proposal.wintercg.org/#edge-light
 		conditions: ['edge-light', 'worker', 'browser'],
-		external: ['astro/middleware'],
 		outfile: bundledFilePath,
 		allowOverwrite: true,
 		format: 'esm',
 		bundle: true,
 		minify: false,
+		plugins: [{
+			name: 'esbuild-namespace-node-built-in-modules',
+			setup(build) {
+				const filter = new RegExp(builtinModules.map((mod) => `(^${mod}$)`).join('|'));
+				build.onResolve({ filter }, (args) => ({ path: 'node:' + args.path, external: true }));
+			},
+		}]
```

Comment:
Could you add a comment that explains this plugin and what it's for?

Reply:
I can see why. Wouldn't this affect the runtime, though? Here, we use the user's Node.js version to create the edge function. This means that there could be a situation where the user's node.js version will be different from the edge middleware version.

What happens if a user uses a module in Node.js v20 but not in Node.js v18 - which is the version of the edge middleware?

Reply:
That is a real corner case. Currently, `inspector/promises` is the one module absent from node 18 but available in node 20. I'm guessing the error message from vercel in that case would point the users towards the right node version they should configure.